connections/README.md

```diff
@@ -328,6 +350,45 @@ baseline would be:
 | Listen       | We've started listening on a new address  |
 | ListenClose  | We've stopped listening on an address     |
 
+## Future Work
+
+A replacement for multistream-select is [being discussed][mss-2-pr] which
+proposes solutions for several inefficiencies and shortcomings in the current
+protocol negotiation and connection establishment process. The ideal outcome of
+that discussion will require many changes to this document, once the new
+multistream semantics are fully specified.
+
+For connection management, there is currently a draft of a [connection manager
+specification][connmgr-v2-spec] that may replace the current [connmgr
+interface][connmgr-go-interface] in go-libp2p and may also form the basis of
+other connection manager implementations. There is also [a
+proposal][resource-manager-issue] for a more comprehensive resource management
+system, which would track and manage other finite resources as well as
+connections.
+
+Also related to connection management, libp2p has recently added support for
+[QUIC][quic-spec], a transport protocol layered on UDP that can resume sessions
+with much lower overhead than killing and re-establishing a TCP connection. As
+QUIC and other "connectionless" transports become more widespread, we want to
+take advantage of this behavior where possible and integrate lightweight session
+resumption into the connection manager.
+
+Event delivery is also undergoing a refactoring in go-libp2p, with work on an
+[in-process event bus][go-eventbus] in progress now that will augment (and
+perhaps eventually replace) the current [notification system][go-net-notifee].
+
+One of the near-term goals of the event bus refactor is to more easily respond
+to changes in the protocols supported by a remote peer. Those changes are
+communicated over the wire using the [identify/push protocol][identify-push].
+Using an event bus allows other, unrelated components of libp2p (for
+example, a DHT module) to respond to changes without tightly coupling components
+together with direct dependencies.
+
+While the event bus refactoring is specific to go-libp2p, a future spec may
+standardize event types used to communicate information across key libp2p
+subsystems, and may possibly require libp2p implementations to provide an
```

Comment:
Fwiw, that's something that rust-libp2p would definitely never implement.

Reply:
Good to know 😄

I honestly don't know if there's value in specifying the event system, since it feels like an implementation detail. The reason I added this note is that there's some ongoing work now (in go-libp2p) to dynamically change the DHT behavior in response to events from another protocol (identify). It seems likely that other kinds of "loosely coupled" interactions are likely to emerge in go-libp2p once the event bus is in place. So really I just want something to refer to when writing about those kinds of interactions, e.g. "the DHT may do X in response to Y events from component Z". 

I think actually spec'ing the mechanism for event delivery is probably too much, and maybe even referring to "events" at all is too implementation-specific.